VideoAPI/Video.API/Controllers/ConsultationController.cs

```diff
@@ -240,6 +246,55 @@ public async Task<IActionResult> StartPrivateConsultationWithEndpointAsync(Endpo
 
             return Accepted();
         }
+
+        [HttpPost("/consultations/start")]
+        [SwaggerOperation(OperationId = "StartPrivateConsultation")]
+        [ProducesResponseType((int) HttpStatusCode.Accepted)]
+        [ProducesResponseType((int) HttpStatusCode.NotFound)]
+        [ProducesResponseType((int) HttpStatusCode.Unauthorized)]
+        [ProducesResponseType((int) HttpStatusCode.BadRequest)]
+        [AllowAnonymous]
+        public async Task<IActionResult> StartConsultationRequestAsync(StartConsultationRequest request)
+        {
+            _logger.LogDebug("StartJudgeJohConsultationRequest");
+            Room room;
+            Participant participant;
+            
+            try
+            {
+                var query = new GetAvailableRoomByRoomTypeQuery(request.RoomType, request.ConferenceId);
+                var listOfRooms = await _queryHandler.Handle<GetAvailableRoomByRoomTypeQuery, List<Room>>(query);
+                room = listOfRooms.FirstOrDefault(x => x.Type.Equals(request.RoomType));
+                if (room == null)
+                {
+                    var createConsultationRoomResponse = await CreateNewConsultationRoom(request);
+                    room = new Room(request.ConferenceId, createConsultationRoomResponse.Room_label, request.RoomType);
+                }
+            }
+            catch (ConferenceNotFoundException e)
+            {
+                _logger.LogError("Cannot create consultation for conference: {conferenceId} as the conference does not exist",
+                    request.ConferenceId);
+                return NotFound(e.Message);
+            }
+
+            try
+            {
+                var conference = await _queryHandler.Handle<GetConferenceByIdQuery, Conference>(new GetConferenceByIdQuery(request.ConferenceId));
+                participant = conference.GetParticipants().Single(x => x.Id == request.RequestedBy);
+            }
+            catch (ParticipantNotFoundException e)
+            {
+                _logger.LogError("Cannot create consultation with participant: {participantId} as the conference does not exist",
+                    request.RequestedBy);
+                return NotFound(e.Message);
+            }
+
+            await _consultationService.TransferParticipantAsync(request.ConferenceId, request.RequestedBy, participant.CurrentRoom.ToString(), room.Label);
+            _logger.LogTrace("{ParticipantId} successfully transferred in consultation room: {roomLabel}", request.RequestedBy, room.Label);
+
+            return Accepted();
```

Comment:
Should we return Room Id, as it required in the story that calls this endpoint?

Reply:
We don't need to return a room id here since we're only supporting one room type, and that room type will either be live or closed.

Reply:
in the story:  4.Return Room id returned by the video API. This will be used to transfer people out of the room at the end of the consultation